packages/env/lib/env.js

```diff
@@ -469,28 +468,34 @@ async function configureWordPress( environment, config ) {
 		if ( typeof value !== 'string' ) {
 			command.push( '--raw' );
 		}
-		await dockerCompose.run(
-			environment === 'development' ? 'cli' : 'tests-cli',
-			command,
-			options
-		);
+		await dockerCompose.run( containerName, command, options );
 	}
 
 	// Activate all plugins.
-	for ( const pluginSource of config.pluginSources ) {
-		await dockerCompose.run(
-			environment === 'development' ? 'cli' : 'tests-cli',
-			`wp plugin activate ${ pluginSource.basename }`,
-			options
-		);
+	await dockerCompose.run(
+		containerName,
+		'wp plugin activate --all',
```

Comment:
I just realised that this approach of activating all plugins using `wp plugin activate --all` will mean that all of the Gutenberg test plugins (`packages/e2e-tests/plugins`) will be activated by default. That's probably not what we want for the default Gutenberg development environment.

I'm not sure of the best way around this. Some ideas:

1. We could look at adding support for plugins that are not activated automatically.
2. We could look at adding support for plugins that are only loaded into the tests environment. 
3. We could go back to hardcoding this for Gutenberg.

Reply:
That's an interesting conundrum. My intuition would be to do number 2:

- number 3 isn't an ideal choice any way you cut it
- number 1 means we wouldn't directly support test environment mu-plugins. I assume we want to automatically activate them for the test environment :)

So for the api, maybe we change the source format in wp-env to accept `string | object` where the string is a normal source string (current approach) and the object has the format:

```js
{
  "source": SourceString,
  "activateInEnv": "development" | "tests" | "none"
}
```

Then, we activate the plugin depending on the environment, or don't activate it at all if none is passed. (just an idea on how to make this a bit more flexible in the long run).

BTW, will this approach work out of the box with the current docker setup? i.e. does `test-cli wp activate plugin` only activate the plugin on the test WP and not on the production one?

Reply:
> * I assume we want to automatically activate them for the test environment :)

The E2E tests themselves are responsible for activating the plugins that they need, e.g:

https://github.com/WordPress/gutenberg/blob/ea62337a254c52305ee6bb46e00656d665123cc3/packages/e2e-tests/specs/editor/plugins/container-blocks.test.js#L15-L17

> So for the api, maybe we change the source format in wp-env to accept string | object where the string is a normal source string (current approach) and the object has the format:

I'm not a huge fan on the extra weight this adds to `.wp-env.json`. I worry about making this file so complex that it doesn't provide much benefit over using `docker-compose.yml`.

> BTW, will this approach work out of the box with the current docker setup? i.e. does test-cli wp activate plugin only activate the plugin on the test WP and not on the production one?

Yes, that will work because the instances use different databases.

Reply:
I'm leaning towards option one because it maintains one principal of `wp-env` that I like which is that the two instances it spins up are identical: same core code, same plugins, same themes. It's really nice being able to make this assumption when e.g. writing and debugging E2E tests.

In other words, I think let's add some way of defining a plugin that isn't activated, similar to what you suggested above:

```json
{
	"plugins": [
		".",
		{ "source": "./packages/e2e-tests/plugins", "active": false }
	]
}
```

This is tough to implement, though. We can't use `wp plugin activate --all` since that will activate everything. We also can't iterate through the plugins with `.active = true` and perform `wp plugin activate ${plugin}` because some of the paths will be to _directories containing plugins_ instead of to plugins themselves. We might have to do some weird stuff involving `wp plugin list`, `wp plugin path`, and `wp plugin activate`.

😕

Reply:
I like that idea. For example, active could default to true. I'm exploring this in another branch -- I think it's a good opportunity to make sure directory support works in general too.

Reply:
what about mu-plugins that only need to be activated in the test environment?

Reply:
https://github.com/WordPress/gutenberg/pull/21710